### ISSUE TITLE
(ec2.CfnVPC): Missing AWS IPAM properties

### ISSUE THREAD

**Issue by vahagn-madatyan** [aws/aws-cdk#18968]
### What is the problem?

The [CfnVPC module](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_ec2/CfnVPC.html) is missing available properties for AWS IPAM(Ipv4IpamPoolId & Ipv4NetmaskLength) as described [here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html)

While you can use escape hatches to add property override, this does not work, since cdk library is coded to require "cidr_block" property.

This was a bug in cloudformation that is now fixed but not reflected in AWS CDK.

### Reproduction Steps

1. Add cfnVPC
`vpc_test = ec2.CfnVPC(
            self,
            "test-vpc",
            enable_dns_hostnames=True,
            enable_dns_support=True,
        )`
2. Add property override(since its missing in CDK module)
`vpc_test.add_property_override("Ipv4IpamPoolId", "ipam-pool-XXX")
vpc_test.add_property_override("Ipv4NetmaskLength", 20)`
3. CDK Deploy
4.
![image](https://user-images.githubusercontent.com/74753362/153934963-40212704-77f4-4dd1-b41f-2b7ae10b1d5f.png)

### What did you expect to happen?

CDK should be able to at minimum deploy CfnVPC with property override and ideally also have the properties added

### What actually happened?

![image](https://user-images.githubusercontent.com/74753362/153935096-70527e06-8009-4c99-aded-d42d29399283.png)

### CDK CLI Version

2.12.0 (build c9786db)

### Node.js Version

v14.18.0

### OS

MacOS Monterey

### Language

Python

### Language Version

Python 3.9.7

### Other information

Suggestion for fix:

1. Make cidr_block property optional with default value of some CIDR(10.0.0.0/16). 
2. If IPAM properties are given, use those. 
3. If cidr_block is specified, use those.
4. If cidr_block & IPAM is specified, throw error
5. If none are specified, use cidr_block default

**Comment by peterwoodworth** [comment]
our CloudFormation resources are automatically generated from the CloudFormation [resource spec](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-resource-specification.html). There's usually a delay of a week or two until we merge the changes.

Another side effect of our L1 resources being generated (prefixed with `Cfn`) is that we both can't, and don't want to provide anything different than exactly what CloudFormation provides. [Cloudformation requires the cidrBlock property](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-ec2-vpc.html#cfn-aws-ec2-vpc-cidrblock), so CfnVpc will as well.

**Comment by vahagn-madatyan** [comment]
@peterwoodworth  I see what you are saying. Cloudformation backend is actually updated to not require cidr block, it was a bug that was addressed after introduction of IPAM. I was able to vend a VPC via cloudformation without the cidr block property. I suppose they did not update the spec files tho. Thank you!

**Comment by rkolias** [comment]
I am suffering from this one from the absence of the Ipv4IpamPoolId property, which is now implemented in CloudFormation (though very recently). 

What threw me is the v2.14.0 documentation shows this property as present, however tracing the git subway map shows that commit (24a52ae1c250ec1875e64d6fc4ef8bec2f47399a) seems not to be recognized when I build my project (no such property). I do see it in the source however. 

https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2.CfnVPCProps.html

**Comment by peterwoodworth** [comment]
I'm not finding the same behavior @rkolias, the new properties `ipv4IpamPoolId` and `Ipv4NetmaskLength` exist for me on `CfnVPC`. are you sure your locally installed packages are on the latest version?

**Comment by rkolias** [comment]
Hi @peterwoodworth. Yah, pretty sure. See attached screen shot. Note here v2.14.0, but both building and autocomplete in VS Code both indicate those two props are absent. Honestly I would be delighted if I am just doing something dumb.

What else can I attempt?

![image](https://user-images.githubusercontent.com/1790770/156233666-92cad954-e1e3-4901-be39-d8ff0472720b.png)

**Comment by peterwoodworth** [comment]
Have you tried removed your packages and reinstalled them? Just having your CLI be the latest version doesn't ensure that your node packages are updated
```
rm -rf package-lock.json node_modules
npm i
```

**Comment by peterwoodworth** [comment]
Oh, and check your package.json too to make sure that the version of the cdk packages is `2.14.0`

**Comment by rkolias** [comment]
@peterwoodworth Thank you for taking the time to comment. So, I think solved. Here appears to be the causes:

1) It seems the install/init commands suggested in the 'Getting started' portion of the CDK Developer Guide installed the CDK version older than the commit above (which seems to be have been a couple weeks ago).

2) Only the level 1 CfnVpc construct contains the new attrbs. the "Vpc" construct does not. Coding directly to CfnVpc (with CfnVpcProps), and reinstalled the CDK above, I was able to reference the IPAM attribs.

Not sure about the OP, but for me this issue is resolved. Def thanks @peterwoodworth  for the prod in the right direction.

**Comment by peterwoodworth** [comment]
I forgot about the `cidrBlock` prop here, it is optional even though according to the resource spec it says it's required. When creating a CfnVpc you will still have to fill out the CidrBlock property, but you can easily remove this on the final resource with `addPropertyDeletionOverride()`

There's an open ticket internally for CFN to fix this, once that goes through then it shouldn't take more than a couple weeks for the changes to be reflected in our library

**Comment by rkolias** [comment]
@peterwoodworth Yes, I hit that barrier right away. CDK mandates the cidrBlock property on CfnVpc, but of course the entire point of IPAM is that the cidr block is allocated dynamically from available blocks in the pool. Thanks for picking up on that.

I will try calling `addPropertyDeletionOverride()` as suggested.
